adapters/src/main/java/io/realm/RealmCursor.java

```diff
@@ -0,0 +1,688 @@
+            default:
+                throw canNotConvert(currentRow, columnIndex, "byte[]");
+        }
+    }
+
+    @Override
+    @Nullable
+    public String getString(int columnIndex) {
+        Row currentRow = getCurrentRow();
+        checkValidColumnIndex(columnIndex);
+        RealmFieldType columnType = currentRow.getColumnType(columnIndex);
+
+        if (currentRow.isNull(columnIndex)) {
+            return null;
+        }
+
+        switch (columnType) {
+            case STRING:
+                return currentRow.getString(columnIndex);
+            case INTEGER:
+                return Long.toString(currentRow.getLong(columnIndex));
+            case FLOAT:
+                return Float.toString(currentRow.getFloat(columnIndex));
+            case DOUBLE:
+                return Double.toString(currentRow.getDouble(columnIndex));
+            case BOOLEAN:
+                return Boolean.toString(currentRow.getBoolean(columnIndex));
+            case DATE:
+                return currentRow.getDate(columnIndex).toString();
+            default:
+                throw canNotConvert(currentRow, columnIndex, "String");
+        }
+    }
+
+
+    @Override
+    public void copyStringToBuffer(int columnIndex, @NonNull CharArrayBuffer buffer) {
+        String string = getString(columnIndex);
+
+        if (string == null) {
+            buffer.sizeCopied = 0;
+            return;
+        }
+
+        if (buffer.data.length < string.length()) {
+            char[] chars = string.toCharArray();
+            buffer.data = chars;
+            buffer.sizeCopied = chars.length;
+            return;
+        }
+
+        string.getChars(0, string.length(), buffer.data, 0);
+        buffer.sizeCopied = string.length();
+    }
+
+    @Override
+    public short getShort(int columnIndex) {
+        Row currentRow = getCurrentRow();
+        checkValidColumnIndex(columnIndex);
+        RealmFieldType columnType = currentRow.getColumnType(columnIndex);
+
+        switch (columnType) {
+            case INTEGER:
+                long aLong = currentRow.getLong(columnIndex);
+                if (aLong > Short.MIN_VALUE && aLong < Short.MAX_VALUE) {
+                    return (short) aLong;
+                }
+            case BOOLEAN:
+                return currentRow.getBoolean(columnIndex) ? (short) 1 : (short) 0;
+            default:
+                throw canNotConvert(currentRow, columnIndex, "short");
+        }
+    }
+
+    @Override
+    public int getInt(int columnIndex) {
+        Row currentRow = getCurrentRow();
+        checkValidColumnIndex(columnIndex);
+        RealmFieldType columnType = currentRow.getColumnType(columnIndex);
+
+        switch (columnType) {
+            case INTEGER:
+                long aLong = currentRow.getLong(columnIndex);
+                if (aLong > Integer.MIN_VALUE && aLong < Integer.MAX_VALUE) {
+                    return (int) aLong;
+                }
```

Comment:
If `aLong` is out of range here, the caller should have used `getLong(...)` instead of `getInt(...)`